packages/next/src/server/lib/start-server.ts

```diff
@@ -327,7 +336,7 @@ export async function startServer(
           dir,
           port,
           isDev,
-          onCleanup: (listener) => cleanupListeners.push(listener),
+          onCleanup,
```

Comment:
Looking into the code, this is only called and added for development. This shouldn't change the code, but thought it important to note that the code here is a bit convoluted. Maybe we should rename this `onCleanup` here specifically to `onDevCleanup` and then restore the old pattern using your method of:

```suggestion
          onDevCleanup: (listener) => onCleanup(listener),
```

Reply:
yeah good call, i got confused by this again while reading this suggestion, so others definitely would as well. renamed that one to `onDevServerCleanup` and made it so that we only pass it in during dev.